File: lib/result/Rssi.java
package com.teamlab.communication.wireless.lpr9201.result;

/**
 * リザルト Rssi
 */
public class Rssi extends Result {
	/**
	 * 
	 */
	public static final int RESULT_CODE = 0x84;

	/**
	 * 
	 */
	public static final int DATA_LENGTH_SIZE = 1; // byte
	
	private int value;

	public Rssi(int[] datas) {
		super(RESULT_CODE, DATA_LENGTH_SIZE, datas);
		this.value = datas.length > 0 ? datas[0] : 0;
	}

	public int getValue() {
		return this.value;
	}
}

File: lib/result/rssi.js
/* jslint node: true */
"use strict"

/**
 * リザルト Rssi
 */

exports.Rssi = Rssi;

/**
 * 結果コード
 */
Rssi.RESULT_CODE = 0x84;

/**
 * データ長バイトサイズ
 */
Rssi.DATA_LENGTH_BYTE_SIZE = 1; // byte


/**
 * @param {Result} result
 */
function Rssi(result) {
  this.result = result;
  this.value = result.datas.length > 0 ? result.datas[0] : 0;
}

/**
 * パース可能か
 * @param {Result} result
 */
Rssi.canParse = function(result) {
  return result !== null && result.resultCode === Rssi.RESULT_CODE;
}
